samples/create_vm.py

```diff
@@ -0,0 +1,184 @@
+
+    parser.add_argument('-g', '--guest_type',
+                        required=True,
+                        action='store',
+                        help='VM guest type')
+
+    parser.add_argument('-d', '--datastore',
+                        required=True,
+                        action='store',
+                        help='Name of Datastore to create VM in')
+
+    parser.add_argument('--datacenter',
+                        required=True,
+                        action='store',
+                        help='Name of data center to create VM in')
+
+    parser.add_argument('-r', '--remote_file',
+                        required=True,
+                        action='store',
+                        help='Path on datastore where vmdk is')
+
+    args = parser.parse_args()
+
+    return cli.prompt_for_password(args)
+
+
+def find_vm(service_instance, name):
+    obj = None
+
+    content = service_instance.RetrieveContent()
+    container = content.viewManager.CreateContainerView(
+        content.rootFolder, [vim.VirtualMachine], True)
+    for c in container.view:
+        if c.name == name:
+            obj = c
+            break
+    return obj
+
+
+def create_vm(vm_name, vmdk_path, vm_ram, vm_guest,
+              service_instance, data_store, data_center,
+              vm_folder, resource_pool):
+    """Creates a VirtualMachine.
+
+    :param vm_name: String Name for the VirtualMachine
+    :param vmdk_path: Full path to vmdk file in the datastore
+    :param vm_ram: Memory in MB
+    :param vm_guest: Guest type, e.g. rhel7_64Guest
+    :param service_instance: ServiceInstance connection
+    :param data_store: DataStore to place the VirtualMachine on
+    :param data_center: DataCenter
+    :param vm_folder: Folder to place the VirtualMachine in
+    :param resource_pool: ResourcePool to place the VirtualMachine in
+    """
+    # bare minimum VM shell, no disks.
+    vmx_path = '[' + data_store.info.name + '] ' + vm_name
+    vmx_file = vim.vm.FileInfo(logDirectory=None,
+                               snapshotDirectory=None,
+                               suspendDirectory=None,
+                               vmPathName=vmx_path)
+
+    # config for VM
+    config = vim.vm.ConfigSpec(name=vm_name, memoryMB=vm_ram, numCPUs=1,
+                               files=vmx_file, guestId=vm_guest,
+                               version='vmx-13')
+
+    print("Creating VM {}...".format(vm_name))
+    task = vm_folder.CreateVM_Task(config=config, pool=resource_pool)
+    tasks.wait_for_tasks(service_instance, [task])
+
+    vm = find_vm(service_instance, vm_name)
+
+    ### good till here
+    device=vim.vm.device.VirtualDisk(
+        controllerKey=None,
+        backing=vim.vm.device.VirtualDisk.FlatVer2BackingInfo(
+            datastore=data_store,
+            fileName=vmdk_path,
+        ),
+    )
+
+    disk_spec = vim.vm.device.VirtualDeviceSpec(
+        device=device,
+        fileOperation="create",
```

Comment:
You should remove this line as it indicates to create a new file.
If it already exists then you don't need a fileOperation.